runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolServiceTest.java

```diff
@@ -120,4 +121,16 @@ public void onCompleted() {
     latch.await();
     assertThat(sawComplete.get(), is(true));
   }
+
+  @Test
+  public void testUnknownBundle() throws Exception {
+    BeamFnApi.GetProcessBundleDescriptorRequest request =
+        BeamFnApi.GetProcessBundleDescriptorRequest.newBuilder()
+            .setProcessBundleDescriptorId("missing")
+            .build();
+    StreamObserver<BeamFnApi.ProcessBundleDescriptor> responseObserver = mock(StreamObserver.class);
+    controlService.getProcessBundleDescriptor(request, responseObserver);
+
+    verify(responseObserver).onError(any(StatusException.class));
```

Comment:
Validate the "not found" shows up in the status message and/or status code to ensure that we got the right StatusException